helix-rest/src/main/java/org/apache/helix/rest/server/ServerContext.java

```diff
@@ -157,6 +158,50 @@ public RealmAwareZkClient getRealmAwareZkClient() {
     return _zkClient;
   }
 
+  /**
+   * Returns a RealmAWareZkClient with ByteArraySerializer with double-checked locking.
+   * NOTE: this is different from getRealmAwareZkClient in that it does not reset listeners for
+   * _zkClientForListener because this RealmAwareZkClient is independent from routing data changes.
+   * @return
+   */
+  public RealmAwareZkClient getByteArrayRealmAwareZkClient() {
```

Comment:
Just have one suggestion. There are many duplicated pieces with previous function. Can we abstract it as a private function and called in both public method?

Reply:
+1, the only difference is the serializer, right?

Reply:
I agree with the suggestions made here. This PR has been updated with the refactor.